Editor/Window/UserProfileSelection.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Linq;
+using System.Threading;
+using Editor.CoreAPI;
+using UnityEngine.UIElements;
+
+namespace AmazonGameLift.Editor
+{
+    internal class UserProfileSelection
+    {
+        private BootstrapSettings _bootstrapSettings;
+        private CancellationTokenSource _refreshBucketsCancellation;
+        private readonly VisualElement _container;
+        private readonly AwsUserProfilesPage _awsUserProfilesPage;
+        private readonly StateManager _stateManager;
+        
+        public UserProfileSelection(VisualElement container, StateManager stateManager, AwsUserProfilesPage profilesPage)
+        {
+            _container = container;
+            _stateManager = stateManager;
+            _awsUserProfilesPage = profilesPage;
+            _bootstrapSettings = profilesPage.BootstrapSettings;
+        }
+
+        private void BucketSelection()
```

Comment:
Please rename to `SelectBucket` since that's the convention we typically follow internally

Reply:
Renamed